codegen/service/endpoint.go

```diff
@@ -184,10 +184,14 @@ type {{ .VarName }} struct {
 
 // input: EndpointsData
 const serviceEndpointsInitT = `{{ printf "New%s wraps the methods of the %q service with endpoints." .VarName .Name | comment }}
-func New{{ .VarName }}(s {{ .ServiceVarName }}{{ range .Schemes }}, auth{{ . }}Fn security.Auth{{ . }}Func{{ end }}) *{{ .VarName }} {
+func New{{ .VarName }}(s {{ .ServiceVarName }}) *{{ .VarName }} {
+{{- if .Schemes }}
+	// Typing service to authenticated interface
+	a := s.(authenticated)
```

Comment:
As per the comment below:
```suggestion
	a := s.(auther)
```